File: CCC2000/CCC00S1.java
import java.util.Scanner;

public class CCC00S1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int quarters = sc.nextInt();
        int a = sc.nextInt();
        int b = sc.nextInt();
        int c = sc.nextInt();

        int plays = 0;

        while (quarters > 0) {
            // Machine A
            quarters--;
            plays++;
            a++;
            if (a == 35) {
                quarters += 30;
                a = 0;
            }
            if (quarters == 0)
                break;

            // Machine B
            quarters--;
            plays++;
            b++;
            if (b == 100) {
                quarters += 60;
                b = 0;
            }
            if (quarters == 0)
                break;

            // Machine C
            quarters--;
            plays++;
            c++;
            if (c == 10) {
                quarters += 9;
                c = 0;
            }
        }

        System.out.println("Martha plays " + plays + " times before going broke.");
    }
}

File: CCC2000/CCC00S1.cpp
#include <iostream>
using namespace std;

int main() {
    int q, a, b, c, p = 0;
    
    cin >> q >> a >> b >> c;

    while (q > 0) {
        //First machine
        a++;
        q--;
        p++;
        if (a == 35) {
            q += 30;
            a = 0;
        }
        if (q == 0) {
            break;
        }

        //Second machine
        b++;
        q--;
        p++;
        if (b == 100) {
            q += 60;
            b = 0;
        }
        if (q == 0) {
            break;
        }

        //Third machine
        c++;
        q--;
        p++;
        if (c == 10) {
            q += 9;
            c = 0;
        }
        if (q == 0) {
            break;
        }
    }
    cout << "Martha plays " << p << " times before going broke.";
}
